src/js/components/cycleCount/tableCell/CommentCell.jsx

```diff
@@ -0,0 +1,67 @@
+import React, { useEffect, useMemo, useState } from 'react';
+
+import { useDispatch, useSelector } from 'react-redux';
+import { makeGetCycleCountItem } from 'selectors';
+
+import { updateFieldValue } from 'actions';
+import { TableCell } from 'components/DataTable';
+import InputField from 'components/form-elements/v2/TextInput';
+
+const CommentCell = ({
+  id,
+  cycleCountId,
+  isStepEditable,
+}) => {
+  const getCycleCountItem = useMemo(() => makeGetCycleCountItem(), []);
+  const item = useSelector(
+    (state) => getCycleCountItem(state, cycleCountId, id),
+  );
+  const { comment: initialValue } = item;
+
+  const [value, setValue] = useState(initialValue);
+
+  useEffect(() => {
+    setValue(initialValue);
+  }, [initialValue]);
+
+  if (!isStepEditable) {
+    return (
+      <TableCell
+        className="static-cell-count-step d-flex align-items-center"
+      >
+        {value}
+      </TableCell>
+    );
+  }
+
+  const dispatch = useDispatch();
+
+  const onChange = (e) => {
+    setValue(e.target.value);
+  };
+
+  const onBlur = () => {
+    dispatch(
+      updateFieldValue({
+        cycleCountId,
+        rowId: id,
+        field: 'comment',
+        value,
+      }),
+    );
+  };
```

Comment:
looking at another similar cell compoments, it feels like we could wrap some logic into a custom hook - see that you have repeated a lot of the code - `useEffect`, value state, onChange method, onBlur method.
Also a question - doesn't it affect the flow of our react-hook-form? I'm asking because it feels weird that we have to write `onChange` methods manually here, but maybe I'm missing the context.

Reply:
We don't use react hook form here